mslib/msui/mscolab.py

```diff
@@ -693,7 +697,35 @@ def after_login(self, emailid, url, r):
 
             self.signal_login_mscolab.emit(self.mscolab_server_url, self.token)
 
-    def fetch_gravatar(self, refresh=False):
+    def set_profile_pixmap(self, img_data):
+        pixmap = QPixmap()
+        pixmap.loadFromData(img_data)
+        resized_pixmap = pixmap.scaled(64, 64)
+
+        if (hasattr(self, 'profile_dialog') and self.profile_dialog is not None and
+                hasattr(self.profile_dialog, 'gravatarLabel') and self.profile_dialog.gravatarLabel is not None):
+            self.profile_dialog.gravatarLabel.setPixmap(resized_pixmap)
+
+        if hasattr(self, 'ui'):
+            icon = QtGui.QIcon()
```

Comment:
likly this hasattr has also some yet hidden problems.

Reply:
when is it not True?

Reply:
my bad, it is truthy

Reply:
I think I would have done it as a preventive measure